zb-db/src/test/java/io/camunda/zeebe/db/impl/rocksdb/transaction/ForeignKeyCheckerTest.java

```diff
@@ -63,6 +66,71 @@ void shouldSucceedOnExistingForeignKey() throws Exception {
     check.assertExists(tx, new DbForeignKey<>(key, TestColumnFamilies.TEST_COLUMN_FAMILY));
   }
 
+  @Test
+  void shouldRespectSkipCondition() throws Exception {
+    // given
+    final var db = mock(ZeebeTransactionDb.class);
+    final var tx = mock(ZeebeTransaction.class);
+    final var check = new ForeignKeyChecker(db, new ConsistencyChecksSettings(true, true));
+    final var key = new DbLong();
+
+    // when -- tx says no key exists
+    when(tx.get(anyLong(), anyLong(), any(), anyInt())).thenReturn(null);
+
+    // then
+    assertDoesNotThrow(
+        () -> {
+          key.wrapLong(-1);
+          check.assertExists(
+              tx,
+              new DbForeignKey<>(
+                  key,
+                  TestColumnFamilies.TEST_COLUMN_FAMILY,
+                  MatchType.Full,
+                  (k) -> k.getValue() == -1));
+        });
+    assertThatThrownBy(
+            () -> {
+              key.wrapLong(5);
+              check.assertExists(
+                  tx,
+                  new DbForeignKey<>(
+                      key,
+                      TestColumnFamilies.TEST_COLUMN_FAMILY,
+                      MatchType.Full,
+                      (k) -> k.getValue() == -1));
+            })
+        .isInstanceOf(ZeebeDbInconsistentException.class)
+        .hasMessageContaining("Foreign key");
+  }
+
+  @Test
+  void shouldCheckIfKeyIsNotSkipped() throws Exception {
+    // given
+    final var db = mock(ZeebeTransactionDb.class);
+    final var tx = mock(ZeebeTransaction.class);
+    final var check = new ForeignKeyChecker(db, new ConsistencyChecksSettings(true, true));
+    final var key = new DbLong();
+
+    // when -- tx says no key exists
+    when(tx.get(anyLong(), anyLong(), any(), anyInt())).thenReturn(null);
+
+    // then
+    assertThatThrownBy(
+            () -> {
+              key.wrapLong(5);
+              check.assertExists(
+                  tx,
+                  new DbForeignKey<>(
+                      key,
+                      TestColumnFamilies.TEST_COLUMN_FAMILY,
+                      MatchType.Full,
+                      (k) -> k.getValue() == -1));
```

Comment:
:x: this would be easier to read if `(k) -> false / true` was used throughout these tests. 

The way it is now, I need to think too much. _So the key is 5, which is not -1. So why is it checking it? Oh right, the condition was for when to skip, not when to check_